MGDAPI-1037 Add RBAC markers

sergioifg94 opened the pull request:
# Description

> Link to JIRA: https://issues.redhat.com/browse/MGDAPI-1037

`operator-sdk` uses now RBAC markers to generate the operator `Role` and `ClusterRole` . This change adds the markers to the controllers which generate the correct permissions for the operator

> Information about RBAC markers: https://book.kubebuilder.io/reference/markers/rbac.html

## Notes

## 1. `Role` generation

By default RBAC markers generate `ClusterRoles`, in order to generate a `Role`, a namespace must be specified in the marker. As the operator currently supports dynamic namespaces (either `redhat-rhmi-operator` or `redhat-rhoam-operator`), the markers are currently duplicated, so two `Roles` are generated in the different namespaces

# Verification steps

1. Clone the repo from this PR
2. Run `make manifests`
    > The command will currently fail, but the RBAC will be generated
3. Verify that the permissions listed in `config/rbac/role.yaml` match the permissions existing in [master](https://github.com/integr8ly/integreatly-operator/blob/master/deploy/role.yaml) (considering note 1)

sergioifg94 committed:
chore: Add markers for operator RBAC

Add markers to controllers that generate the correct RBAC for the
operator

obrienrobert commented:
@sergioifg94 just wondering if the likes of [this](https://github.com/integr8ly/integreatly-operator/blob/master/deploy/role.yaml#L473-L479) permission will have to be added manually to the `role.yaml` file, or can it also be auto-generated? I can't seem to find it.

sergioifg94 commented:
@obrienrobert It can be auto-generated as well. My local clone is not up to date so I missed that rule. I'll update the PR

sergioifg94 commented:
@obrienrobert I rebased the migration branch and this one and introduced that missing permission to be generated in the `role.yaml` file:
https://github.com/sergioifg94/integreatly-operator/pull/1/files#diff-6d4c4e29ec096149781536bb9118fde8242c1901df764e3eda3019ac8503d6c7R89-R94

obrienrobert commented:
@sergioifg94 I checked the `config/rbac/role.yaml` against the current master `role.yaml` file and everything looks good to me.
